### ISSUE TITLE
Document error in fields to specify that isInvalid is required.

### ISSUE THREAD

**Issue by JakeLaCombe** [cerner/terra-core#1410]
# Issue Description
While working with terra-form-field, I was attempting to render an error message only to not have it rendered. I looked through the code, and saw it was only rendered if isInvalid is set. I think this approach is correct for the module, but we should document this requirement and any other prop requirements inside terra-form-field.

## Issue Type

- [ ] New Feature
- [x] Enhancement
- [ ] Bug
- [ ] Other

**Comment by saigorantla** [comment]
I am trying to understand this issue, are you expecting the below props document to be updated  as stated below?

current

error | node | optional | null | Error message for when the input is invalid.
-- | -- | -- | -- | --

after enhancement

error | node | optional | null | Error message for when the input is invalid. This message will not be displayed unless isInvalid prop is set to true.
-- | -- | -- | -- | --

Ref:- http://engineering.cerner.com/terra-core/#/site/components/site/form-field/index

**Comment by JakeLaCombe** [comment]
Yup. I would probabaly rephrase it to say `This will only be displayed if isInvalid is true` to save space and condense the message.

**Comment by saigorantla** [comment]
ok I will work on this.
